discovery/consul/consul_test.go

```diff
@@ -46,3 +51,35 @@ func TestNonConfiguredService(t *testing.T) {
 		t.Errorf("Expected service %s to be watched", "nonConfiguredServiceName")
 	}
 }
+
+func TestConsulDiscovery(t *testing.T) {
+	srv1, err := testutil.NewTestServer()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer srv1.Stop()
+
+	srv1.AddService(t, "testService", structs.HealthPassing, []string{"master"})
+	srv1.AddCheck(t, "service:testService", "testService", structs.HealthPassing)
+
+	conf := &config.ConsulSDConfig{
+		Server:       srv1.HTTPAddr,
+		Services:     []string{"testService"},
+		WatchTimeout: 10 * time.Millisecond,
+	}
+	consulDiscovery, err := NewDiscovery(conf)
+	ctx, cancelFunc := context.WithTimeout(context.Background(), time.Millisecond*10)
+	defer cancelFunc()
+	ch := make(chan []*config.TargetGroup, 1)
+
+	consulDiscovery.Run(ctx, ch)
+
+	select {
+	case tg := <-ch:
+		if tg[0].Source != "testService" {
+			t.Errorf("Expected service %s to be watched", tg)
```

Comment:
A slice containing pointers to the actual value, yes. I thought for now, I'd write tests covering the existing behaviour. But I extended it to two services, I think that makes more sense.